From eec523b76adc570f1b7a42b948b3eba03b5594f3 Mon Sep 17 00:00:00 2001
From: Alan Modra <amodra@gmail.com>
Date: Mon, 7 May 2018 22:41:47 +0930
Subject: Bug 23142, SIGSEGV in is_strip_section

	PR 23142
	* objcopy.c (group_signature): Don't accept groups that use a
	symbol table other than the one we've read.
---
 sdcc/support/sdbinutils/binutils/objcopy.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/sdcc/support/sdbinutils/binutils/objcopy.c b/sdcc/support/sdbinutils/binutils/objcopy.c
index 1e39f6d87..f553c55c2 100644
--- a/sdcc/support/sdbinutils/binutils/objcopy.c
+++ b/sdcc/support/sdbinutils/binutils/objcopy.c
@@ -1211,14 +1211,13 @@ group_signature (asection *group)
     return NULL;
 
   ghdr = &elf_section_data (group)->this_hdr;
-  if (ghdr->sh_link < elf_numsections (abfd))
+  if (ghdr->sh_link == elf_onesymtab (abfd))
     {
       const struct elf_backend_data *bed = get_elf_backend_data (abfd);
-      Elf_Internal_Shdr *symhdr = elf_elfsections (abfd) [ghdr->sh_link];
+      Elf_Internal_Shdr *symhdr = &elf_symtab_hdr (abfd);
 
-      if (symhdr->sh_type == SHT_SYMTAB
-	  && ghdr->sh_info > 0
-	  && ghdr->sh_info < (symhdr->sh_size / bed->s->sizeof_sym))
+      if (ghdr->sh_info > 0
+	  && ghdr->sh_info < symhdr->sh_size / bed->s->sizeof_sym)
 	return isympp[ghdr->sh_info - 1];
     }
   return NULL;
-- 
